org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/parser/AbstractCommentParser.java

```diff
@@ -1393,6 +1393,61 @@ protected boolean parseReference() throws InvalidInputException {
 		return parseReference(false);
 	}
 
+	// Parses a complete URL reference starting from current position
+	protected boolean parseURLReference() throws InvalidInputException {
+		int pos = 0;
+		char[]fullURL = null;
+		int firstTokenStartPos;
+		StringBuilder urlBuilder = new StringBuilder();
+		char c;
+		if (this.source[this.index] == '(') {
+			firstTokenStartPos = this.index;
+			while (pos < this.source.length) {
+				c =  readChar();
+				if (c == '(' || c == '[' || c == ' ')
+					continue;
+
+		        if (c == '\t' || c == '\n' || c == '\r' || c == '*' || c == ')' || c == '[' || c == ']' ) {
```

Comment:
Not sure if we should break for anything other than \r and \n.

Reply:
You are right, we do not need most of the conditions. However we may need `c == ')'`, since it calculates the end of the url. `c == ']'` does not required, since the java library references are already addressed in the parseReference method.